src/meta/src/stream/source_manager/split_assignment.rs

```diff
@@ -242,31 +246,27 @@ impl SourceManager {
 
         for (_source_id, fragments) in source_backfill_fragments {
             for (fragment_id, upstream_source_fragment_id) in fragments {
-                let fragment_dispatchers = dispatchers.get(&upstream_source_fragment_id);
-                let upstream_actors = core
-                    .metadata_manager
-                    .get_running_actors_of_fragment(upstream_source_fragment_id)
-                    .await?;
+                let source_upstream_new_no_shuffle =
+                    upstream_new_no_shuffle.get(&upstream_source_fragment_id);
+
+                let upstream_actors = upstream_actors.get(&upstream_source_fragment_id);
                 let mut backfill_actors = vec![];
-                for upstream_actor in upstream_actors {
-                    if let Some(dispatchers) = fragment_dispatchers
-                        .and_then(|dispatchers| dispatchers.get(&upstream_actor))
+                for upstream_actor in upstream_actors.into_iter().flatten() {
+                    if let Some(no_shuffle_backfill_actor) = source_upstream_new_no_shuffle
+                        .and_then(|source_upstream_new_no_shuffle| {
+                            source_upstream_new_no_shuffle.get(&fragment_id)
+                        })
+                        .and_then(|new_no_shuffle| new_no_shuffle.get(upstream_actor))
                     {
-                        let err = || {
-                            anyhow::anyhow!(
-                                "source backfill fragment's upstream fragment should have one dispatcher, fragment_id: {fragment_id}, upstream_fragment_id: {upstream_source_fragment_id}, upstream_actor: {upstream_actor}, dispatchers: {dispatchers:?}",
-                                fragment_id = fragment_id,
-                                upstream_source_fragment_id = upstream_source_fragment_id,
-                                upstream_actor = upstream_actor,
-                                dispatchers = dispatchers
-                            )
-                        };
-                        if dispatchers.len() != 1 || dispatchers[0].downstream_actor_id.len() != 1 {
-                            return Err(err().into());
-                        }
-
-                        backfill_actors
-                            .push((dispatchers[0].downstream_actor_id[0], upstream_actor));
+                        backfill_actors.push((*no_shuffle_backfill_actor, *upstream_actor));
```

Comment:
I'm thinking is `backfill_actors` exactly the same as `upstream_new_no_shuffle[upstream_source_fragment_id][fragment_id]`? Why do we need to iterate over